### PR TITLE
Replace date_sort with descending and ascending options

### PR DESCRIPTION
**Pull request by hackmastera**
Includes a data migration script which should be run on deploy.

refs #2019; close ticket once script has been run

### CODE REVIEW

File: app/services/date_sort_migration.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      exhibit_config.save
+    end
+  end
+
+  def self.re_weight(fields, bump:)
+    return fields unless fields[bump].keys.map(&:to_sym).include?(:weight)
+    weight = fields[bump][:weight]
+    new_weight = (weight.to_i + 1).to_s
+    next_bump = nil
+    fields.each { |k, v| next_bump = k if v[:weight] == new_weight }
```
Comment (tpendragon) [inline app/services/date_sort_migration.rb]: Can this just be
```ruby
next_bump = fields.find{|field_key, field_config| field_config[:weight] == new_weight}&.first?
```
?

File: app/services/date_sort_migration.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      sort_fields["sort_date_desc"][:label] = "Date Descending"
+      sort_fields["sort_date_asc"] = sort_fields["sort_date"].deep_dup
+      sort_fields["sort_date_asc"][:label] = "Date Ascending"
+      sort_fields.delete("sort_date")
+      exhibit_config.sort_fields = re_weight(sort_fields, bump: "sort_date_asc")
+      exhibit_config.save
+    end
+  end
+
+  def self.re_weight(fields, bump:)
```
Comment (tpendragon) [inline app/services/date_sort_migration.rb]: Can we get a comment about why this is necessary?

**Comment by hackmastera** [comment]
@tpendragon ready for re-review. I had to update rubocop to allow the save nav operator; that's done as a separate commit.

**Comment by hackmastera** [comment]
well not update rubocop, update rubocop config.

**Pull request merged**